### PR TITLE
Fix up the build failure due to unused function chooseCore()

### PR DESCRIPTION
**Pull request by jiazhang0**
This commit resolves the following build failure:

src/Arachne.h:460:1: error: ‘int Arachne::chooseCore(const
Arachne::CorePolicy::CoreList&)’ defined but not used
[-Werror=unused-function]
 chooseCore(const CorePolicy::CoreList& coreList) {
 ^~~~~~~~~~

Signed-off-by: Jia Zhang <zhang.jia@linux.alibaba.com>

### CODE REVIEW

File: src/Arachne.h [hunk 1]
```diff
@@ -456,7 +456,7 @@ random(void) {
 // refinement. This function is defined here to facilitate testing; defining it
 // in the CC file causes the compiler to generate an independent version of the
 // random() function above.
-static int
+static int __attribute__ ((unused))
```
Comment (hq6) [inline src/Arachne.h]: @jiazhang0, Which compiler version and which source file is throwing this error? I cannot reproduce this and the function is actually called inside `Arachne.h`, so this error seems incorrect.

Comment (jiazhang0) [reply src/Arachne.h]: @hq6 Here is the error log and g++ version:

In file included from src/Logger.cc:18:0:
src/Arachne.h:460:1: error: ‘int Arachne::chooseCore(const Arachne::CorePolicy::CoreList&)’ defined but not used [-Werror=unused-function]
 chooseCore(const CorePolicy::CoreList& coreList) {
 ^~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [obj/Logger.o] Error 1

$g++ --version
g++ (GCC) 7.2.1 20180104 (Red Hat 7.2.0-5)
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE

I think this fix is cursory.

Comment (hq6) [reply src/Arachne.h]: @jiazhang0 I'm going to get a newer compiler and look into this. We've been compiling on `gcc 4.9.2` with no warnings or errors. I will close this PR once I've fixed this error in a less hacky way.

**Comment by hq6** [comment]
I've pushed this PR manually, since I found a couple of other issues when I compiled using the new compiler that I lumped into one commit.

**Pull request closed without merging**